tsdb/head_test.go

```diff
@@ -1315,3 +1315,74 @@ func TestAddDuplicateLabelName(t *testing.T) {
 	add(labels.Labels{{Name: "a", Value: "c"}, {Name: "a", Value: "c"}}, "a")
 	add(labels.Labels{{Name: "__name__", Value: "up"}, {Name: "job", Value: "prometheus"}, {Name: "le", Value: "500"}, {Name: "le", Value: "400"}, {Name: "unit", Value: "s"}}, "le")
 }
+
+func TestHeadSeriesWithTimeBoundaries(t *testing.T) {
+	h, err := NewHead(nil, nil, nil, 1000, DefaultStripeSize)
+	testutil.Ok(t, err)
+	defer h.Close()
+	app := h.Appender()
+	numSeries := 800
+	for i := 100; i < numSeries; i++ {
+		app.Add(labels.FromStrings(fmt.Sprintf("foo%d", i), "bar"), int64(i), 0)
+	}
+	testutil.Ok(t, app.Commit())
+
+	cases := []struct {
+		mint  int64
+		maxt  int64
+		count int
+	}{
+		{
+			mint:  0,
+			maxt:  1000,
+			count: 700,
+		},
+		{
+			mint:  100,
+			maxt:  799,
+			count: 700,
+		},
+		{
+			mint:  200,
+			maxt:  799,
+			count: 600,
+		},
+		{
+			mint:  100,
+			maxt:  699,
+			count: 600,
+		},
+		{
+			mint:  400,
+			maxt:  499,
+			count: 100,
+		},
+		{
+			mint:  0,
+			maxt:  499,
+			count: 400,
+		},
+		{
+			mint:  500,
+			maxt:  1000,
+			count: 300,
+		},
+	}
+
+	for i, c := range cases {
+		matcher := labels.MustNewMatcher(labels.MatchEqual, "", "")
+		q, err := NewBlockQuerier(h, c.mint, c.maxt)
+		testutil.Ok(t, err)
+
+		count := 0
+		ss, err := q.Select(matcher)
+		testutil.Ok(t, err)
+		for ss.Next() {
```

Comment:
Don't just test the number of series, check the number of samples matches too.

Reply:
done. Added better tests.